consensus/core/src/authority_node.rs

```diff
@@ -145,7 +157,8 @@ where
 pub struct AuthorityService {
     context: Arc<Context>,
     block_verifier: Arc<dyn BlockVerifier>,
-    core_dispatcher: ChannelCoreThreadDispatcher,
+    core_dispatcher: Arc<dyn CoreThreadDispatcher>,
```

Comment:
There may be a conflict with my previous PR, which turned the type here generic. It is ok to use `Arc<dyn CoreThreadDispatcher>`.